lib/in_out/parser.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import re
+
+from parser_regex import log_regex
+
+
+class Parser:
+    def __init__(self, client):
+
+        # self.nicks maintains a list of current user nicks present \
+        # on channels. We will use this heuristic to see nicks mention \
+        # in a message. User mention can happen in following ways: \
+        # <rohit> bhuvan: or <rohit> bhuvan, kausam, or
+        # <rohit> nice work kausam.
+
+        self.nicks = []
+        self.regexes = log_regex[client]
+        # setup logging
+        logging.basicConfig(filename='logger_out.log', level=logging.DEBUG)
+
+    def data_extractor(self, matchObj, groups):
```

Comment:
The purpose of these methods is not clear. A few lines of sphinx-compliant comments for each of the methods would be good.

Reply:
done